src/main/scala/de/frosner/ddq/constraints/ForeignKeyConstraint.scala

```diff
@@ -2,40 +2,79 @@ package de.frosner.ddq.constraints
 
 import org.apache.spark.sql.{Column, DataFrame}
 
+import scala.util.Try
+
 case class ForeignKeyConstraint(columnNames: Seq[(String, String)], referenceTable: DataFrame) extends Constraint {
+
   val fun = (df: DataFrame) => {
     val renamedColumns = columnNames.map{ case (baseColumn, refColumn) => ("b_" + baseColumn, "r_" + refColumn)}
     val (baseColumns, refColumns) = columnNames.unzip
     val (renamedBaseColumns, renamedRefColumns) = renamedColumns.unzip
 
     // check if foreign key is a key in reference table
-    val nonUniqueRows = referenceTable.groupBy(refColumns.map(new Column(_)):_*).count.filter(new Column("count") > 1).count
-    if (nonUniqueRows > 0) {
+    val maybeNonUniqueRows = Try(
+      referenceTable.groupBy(refColumns.map(new Column(_)):_*).count.filter(new Column("count") > 1).count
+    )
+    if (maybeNonUniqueRows.toOption.exists(_ > 0)) {
       ForeignKeyConstraintResult(
         constraint = this,
-        numNonMatchingRefs = None,
+        data = Some(ForeignKeyConstraintResultData(numNonMatchingRefs = None)),
         status = ConstraintFailure
       )
     } else {
       // rename all columns to avoid ambiguous column references
-      val renamedDf = df.select(baseColumns.zip(renamedBaseColumns).map {
+      val maybeRenamedDf = maybeNonUniqueRows.map(_ => df.select(baseColumns.zip(renamedBaseColumns).map {
         case (original, renamed) => new Column(original).as(renamed)
-      }: _*)
-      val renamedRef = referenceTable.select(refColumns.zip(renamedRefColumns).map {
+      }: _*))
+      val maybeRenamedRef = maybeRenamedDf.map(_ => referenceTable.select(refColumns.zip(renamedRefColumns).map {
         case (original, renamed) => new Column(original).as(renamed)
-      }: _*)
+      }: _*))
 
       // check if left outer join yields some null values
-      val leftOuterJoin = renamedDf.distinct.join(renamedRef, renamedColumns.map {
+      val maybeLeftOuterJoin = maybeRenamedRef.map(renamedRef => maybeRenamedDf.get.distinct.join(renamedRef, renamedColumns.map {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of get on optional type.](https://www.codacy.com/app/frank_7/drunken-data-quality/file/2305054514/issues/source?bid=3033737&fileBranchId=3196266#l34)